brain/detail/circuit.h

```diff
@@ -214,16 +197,27 @@ class Circuit::Impl
 
     virtual Vector3fs getPositions(const GIDSet& gids) const = 0;
     virtual size_ts getMTypes(const GIDSet& gids) const = 0;
-    virtual Strings getMorphologyNames() const = 0;
+    virtual Strings getMorphologyTypeNames() const = 0;
     virtual size_ts getETypes(const GIDSet& gids) const = 0;
-    virtual Strings getElectrophysiologyNames() const = 0;
+    virtual Strings getElectrophysiologyTypeNames() const = 0;
     virtual Quaternionfs getRotations(const GIDSet& gids) const = 0;
     virtual Strings getMorphologyNames(const GIDSet& gids) const = 0;
 
     URI getMorphologyURI(const std::string& name) const
     {
+        namespace fs = boost::filesystem;
         URI uri(_morphologySource);
-        uri.setPath(uri.getPath() + "/" + name + ".h5");
+        const auto h5 = uri.getPath() + "/" + name + ".h5";
+        if (!fs::exists(fs::path(h5)))
+        {
+            const auto swc = uri.getPath() + "/" + name + ".swc";
```

Comment:
what happens on the next morphology type we support? :)
Shouldn't this be delegated to the morphology and it's implemented by each plugin there?

Reply:
As discussed off-line, there's no better choice with the current plug-in architecture